src/DynamicData/Cache/Internal/RemoveKeyEnumerator.cs

```diff
@@ -32,14 +40,61 @@ public IEnumerator<Change<TObject>> GetEnumerator()
                 switch (change.Reason)
                 {
                     case ChangeReason.Add:
-                        yield return new Change<TObject>(ListChangeReason.Add, change.Current);
+                        // If not sorted
```

Comment:
There is no need to check for the current index being -1.

```cs
 yield return new Change<TObject>(ListChangeReason.Add, change.Current, change.CurrentIndex);
```

does exactly the same as 

```cs
                        if (change.CurrentIndex == -1)
                        {
                            yield return new Change<TObject>(ListChangeReason.Add, change.Current);
                        }
                        else
                        {
                            yield return new Change<TObject>(ListChangeReason.Add, change.Current, index: change.CurrentIndex);
                        }
```